hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/ConcurrentSchemaEvolutionTableSchemaGetter.java

```diff
@@ -88,10 +82,10 @@ public ConcurrentSchemaEvolutionTableSchemaGetter(HoodieTableMetaClient metaClie
    * @param schema the input schema to process
    * @return the processed schema with partition columns handled appropriately
    */
-  private Schema handlePartitionColumnsIfNeeded(Schema schema) {
+  private HoodieSchema handlePartitionColumnsIfNeeded(HoodieSchema schema) {
     if (metaClient.getTableConfig().shouldDropPartitionColumns()) {
       return metaClient.getTableConfig().getPartitionFields()
-          .map(partitionFields -> appendPartitionColumns(schema, Option.ofNullable(partitionFields)))
+          .map(partitionFields -> TableSchemaResolver.appendPartitionColumns(schema, Option.ofNullable(partitionFields)))
           .or(() -> Option.of(schema))
           .get();
```

Comment:
Nitpick: while we're updating this, let's change this to use `orElseGet` instead of `or` and `get`

Reply:
Done